CommonApi/src/main/java/mezz/jei/api/recipe/category/extensions/IGlobalRecipeCategoryExtension.java

```diff
@@ -0,0 +1,67 @@
+package mezz.jei.api.recipe.category.extensions;
+
+import mezz.jei.api.gui.drawable.IDrawable;
+import mezz.jei.api.gui.ingredient.IRecipeSlotsView;
+import mezz.jei.api.helpers.IGuiHelper;
+import mezz.jei.api.recipe.category.IRecipeCategory;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.network.chat.Component;
+
+import java.util.List;
+
+public interface IGlobalRecipeCategoryExtension<T> {
```

Comment:
It seems a bit dangerous to be able to register one of these for "any recipe category that handles some recipe type".
If we have unknown categories, should we draw on them? I think we might not.
To make this safer, could it be something like `<T, R extends IRecipeCategory<T>>`?

Reply:
I don't really get why it's dangerous to be able to register them, as the user is not able to register an invalid extension to a category anyways. 
What do you mean by unknown category? The extensions can be registered to any kind of category the user wants, so I think we should allow drawing on them.

Reply:
Here's a scenario I have in mind:
Imagine I want to draw something on top of a Thaumcraft recipe where I want to draw a blue circle in the center of the recipe between some ingredients.
I create an extension here and not knowing the difference, I register it for the RecipeType.
I do not know that there is an addon to Thaumcraft that creates a recipe category for this same RecipeType but it is drawn very differently.
Now my extension is drawing on top of that addon's recipes in a way that is broken.

Reply:
Hmm actually I might not know how JEI works.
There can only be one recipe category per recipe type:
https://github.com/mezz/JustEnoughItems/blob/8a65df7e56a1c00d4dd4153260902e4f26a7401a/CommonApi/src/main/java/mezz/jei/api/recipe/RecipeType.java#L9

Reply:
This is fine 👍

Reply:
> Hmm actually I might not know how JEI works.
This made my day 😅